ui/src/features/common/utils.ts

```diff
@@ -28,6 +28,39 @@ export function getCurrentFreight(stage: Stage): FreightReference[] {
     : [];
 }
 
+export function getCurrentFreightWarehouse(stage: Stage) {
+  const freightRef = getCurrentFreight(stage);
+  const isWarehouseKind = freightRef.reduce(
+    (acc, cur) => acc || cur?.origin?.kind === 'Warehouse',
+    false
+  );
```

Comment:
Can you use this? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some